test/abstract_syntax_tree_location_data.coffee

```diff
@@ -1,8 +1,29 @@
-testExpression = expressionAstMatchesObject
+# Astract Syntax Tree location data
+# ---------------------------------
+
+testAstLocationData = (code, expected) ->
+  ast = CoffeeScript.compile code, ast: yes
+  # Pull the node we’re testing out of the root `Block` node’s first child.
+  node = ast.expressions[0]
+
+  eq node.start, expected.start, \
+    "Expected location start #{reset}#{node.start}#{red} to equal #{reset}#{expected.start}#{red}"
+  eq node.end, expected.end, \
+    "Expected location end #{reset}#{node.end}#{red} to equal #{reset}#{expected.end}#{red}"
+  arrayEq node.range, expected.range, \
+    "Expected location range #{reset}#{JSON.stringify node.range}#{red} to equal #{reset}#{JSON.stringify expected.range}#{red}"
+  eq node.loc.start.line, expected.loc.start.line, \
+    "Expected location start line #{reset}#{node.loc.start.line}#{red} to equal #{reset}#{expected.loc.start.line}#{red}"
+  eq node.loc.start.column, expected.loc.start.column, \
+    "Expected location start column #{reset}#{node.loc.start.column}#{red} to equal #{reset}#{expected.loc.start.column}#{red}"
+  eq node.loc.end.line, expected.loc.end.line, \
+    "Expected location end line #{reset}#{node.loc.end.line}#{red} to equal #{reset}#{expected.loc.end.line}#{red}"
+  eq node.loc.end.column, expected.loc.end.column, \
+    "Expected location end column #{reset}#{node.loc.end.column}#{red} to equal #{reset}#{expected.loc.end.column}#{red}"
+
 
 test "AST location data as expected for NumberLiteral node", ->
-  testExpression '42',
-    type: 'NumericLiteral'
```

Comment:
The reason I had the `type`s here was to verify that we're actually examining the node we think we are. The `node = ast.expressions[0]` that is being used to say "we're examining the first thing in this AST" works for now, but won't hold up for long

Eg a real full top-level (file) AST for `42` will end up looking like
```
type: 'File'
program:
  type: 'Program'
  body: [
    type: 'ExpressionStatement'
    expression:
      type: 'NumericLiteral'
  ]
```

Perhaps an elegant mechanism for specify which node the location data belongs to would be to have `testAstLocationData()` accept eg a `path:` argument that describes the path to the desired node? And it could default to the path to the first expression in the `program.body`?

Eg to check the location data for the test of a top-level `if` statement, you could do something like `path: program[0].body.test` (or maybe the initial prefix can be omitted)

Regardless of that (and it's not really necessary yet), I don't see the harm in including a `type:` in these tests, I think all it does is add clarity to what is under test, no?